modules/ROOT/pages/clauses/load-csv.adoc

```diff
@@ -397,4 +570,367 @@ If `LOAD CSV` is invoked with a `file:///` URL that points to your disk `file()`
+
+.Import a CSV with double-quotes escaping
+====
+.artists-with-escaped-quotes.csv
+[source, csv, filename="artists-with-escaped-quotes.csv"]
+----
+"1","The ""Symbol""","1992"
+"1","The \"Symbol\"","1992"
+----
+
+.Query
+[source, cypher]
+----
+LOAD CSV FROM 'file:///artists-with-escaped-quotes.csv' AS row
+MERGE (a:Artist {name: row[1], year: toInteger(row[2])})
+RETURN
+  a.name AS name,
+  a.year AS year,
+  size(a.name) AS size
+----
+
+.Result
+[source, role="queryresult",options="header,footer",cols="3*<m"]
+|===
+| name | year | size
+| 'The "Symbol"' | 1992 | 12
+| 'The "Symbol"' | 1992 | 12
+3+d| Nodes created: 2 +
+Properties set: 4 +
+Labels added: 1
+|===
+
+Note that `name` is a string, as it is wrapped in single quotes in the output.
+The third column outputs the string length as `size`.
+The length only counts what is between the single quotes, but not the quotes themselves.
+
+====
+
+
+=== Check source data quality
+
+In case of a failed import, there are some elements to check to ensure the source file is not corrupted.
+
+- *Inconsistent headers* -- The CSV header may be inconsistent with the data.
+It can be missing, have too many columns or have a different delimiter.
+Verify that the header matches the data in the file.
+Adjust the formatting, delimiters or columns.
+- *Extra or missing quotes* -- Standalone double or single quotes in the middle of non-quoted text or non-escaped quotes in quoted text can cause issues reading the file.
+Either escape or remove stray quotes.
+See <<Quotes escaping>>.
+- *Special or newline characters* -- When dealing with special characters in a file, ensure they are quoted or remove them.
+- *Inconsistent line breaks* -- Ensure line breaks are consistent throughout your file.
+We recommend the Unix style for compatibility with Linux systems. (and is compatible with windows? does it even matter, is not up to the jvm to do the parsing, which is system agnostic?)
+- *Binary zeros, BOM byte order mark and other non-text characters* -- Unusual characters or tool-specific formatting are sometimes hidden in application tools, but become apparent in plain-text editors.
+If you come across these types of characters in your file, remove them entirely.
+
+
+==== Inspect source files ahead of import
+
+Before actually importing data into the database, you can use `LOAD CSV` to inspect a source file and get an idea of what form the imported data is going to have.
+
+.Inspect CSV files with `LOAD CSV`
+====
+[source, cypher]
+----
+// Assert correct line count
+LOAD CSV FROM 'https://data.neo4j.com/importing-cypher/persons.csv' AS line
+RETURN count(*);
+
+// Check first 5 line-sample with header-mapping
+LOAD CSV WITH HEADERS FROM 'https://data.neo4j.com/importing-cypher/persons.csv' AS line
+RETURN line
+LIMIT 5;
+----
+
+.Result
+[source, role="queryresult"]
```

Comment:
This is not the correct return box for the above?

Reply:
true - i'll have another look at returns as a separate step